lib/check-sanity.js

```diff
@@ -0,0 +1,21 @@
+export default function (point) {
+  checkCoord(point.x);
+  checkCoord(point.y);
+}
+function checkCoord(num) {
+  if (typeof Number.isFininte === 'function') {
+    if (Number.isFininte(num)) {
+      return;
+    }
+    throw new TypeError('coordinates must be finite numbers');
+  }
+  if (typeof num !== 'number') {
+    throw new TypeError('coordinates must be finite numbers');
+  }
+  if (num !== num) {
+    throw new TypeError('coordinates must be finite numbers');
+  }
+  if (!isFinite(num)) {
```

Comment:
Number.isFinite checks that it's a number, it's not NaN and that it's not Infinity or -Infinity, the global isFinite function is older, coerces to a number and returns true for NaN.